Fix discussion color contrast and remove gradients

andy-armstrong opened the pull request:
### Description
 
[TNL-5163](https://openedx.atlassian.net/browse/TNL-5163)

This change converts all of the buttons in Discussions over to using Pattern Library styles. This gives them a consistent look, and removes the gradients that made them inaccessible.

### Sandbox
I have a sandbox set up to test the three different discussions:
1. Discussion Board - https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/discussion/forum/
2. Inline Discussion - https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/courseware/social_integration/48ecb924d7fe4b66a230137626bfa93e/
3. Team Discussion - https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/teams/#teams/solar/test-team-385e0cb919c3437481b555543808e7bf

### Testing
- [x] Manual verification of the styling on all three discussion types

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @dianakhuang
- [ ] Code review: @clrux 
- [ ] Doc Review: @catong 
- [x] UX review: @chris-mike 
- [ ] Accessibility review: @clrux 
- [x] Product review: @marcotuts 

### Post-review
- [ ] Squash commits

dianakhuang commented:
It's looking good! Could you remove the color contrast exclusions from the discussion a11y tests as well?

andy-armstrong commented:
@dianakhuang This only fixes the button hover contrast so won't fix any tests. I can pick up the color contrast story too but I won't have time to finish it (I imagine).

dianakhuang commented:
Got it @andy-armstrong . 👍  from me.

chris-mike commented:
Assuming the add a post feature is w/in scope, there are two things that stand out.
First, the "follow this post" "button" seems to not be the default, darker blue. 
Second, the "Topic Area" drop down does not look like it is accessible at the darker grey background.  Is this a button that is w/in scope?  
![screen shot 2016-08-22 at 11 46 36 am](https://cloud.githubusercontent.com/assets/16578372/17861248/41342778-685e-11e6-84b8-1986054214d6.png)

andy-armstrong commented:
Good catches @chris-mike. I've fixed them both and updated my sandbox, so please re-review when you have a chance.

andy-armstrong commented:
@clrux @marcotuts any chance of a quick review today so I can merge before I go on vacation?

chris-mike commented:
👍 Thanks. Enjoy your vacation (as if it needs to be said).

marcotuts commented:
@andy-armstrong - Apologies for missing this yesterday. A few small comments here unfortunately. (Not critical to land during vacation! please ignore!) 

1. Language: Is moving to "Add a Post" intentional? My recommendation would be to use "Add Post" consistently everywhere.
2. CSS Overrides: Text Shadow & Font-Weight are both properties that are being inherited from non-pattern library button styles in the LMS. This is visible in the inline discussion module and teams inline discussion module Add Post buttons. These rules should be overridden so that the buttons look like what the UXPL expects. (Eventually we can clean up / remove all these older terrifying button styles too. :) )

clrux commented:
Not sure if this is in-scope here, but:

* the "0 votes" text is too light
* there is hidden text that only appears on hover (it's opacity is zero) in the same toolset
* the text "Preview" in the preview area has very little contrast

andy-armstrong commented:
@marcotuts The button label change was part of the UX design that went live with the feature branch: https://openedx.atlassian.net/browse/TNL-4522

I'll address the text shadow and font weight properties today.

andy-armstrong commented:
@clrux There are separate accessibility PRs to address the issues you mention. I'll be working on those next.

andy-armstrong commented:
@marcotuts I've fixed the CSS issues with buttons and updated my sandbox. Please re-review when you get a chance.

https://andy-armstrong.sandbox.edx.org/

marcotuts commented:
👍  from me!

andy-armstrong commented:
jenkins run bokchoy
